core/model/modx/processors/system/config_check.class.php

```diff
@@ -0,0 +1,232 @@
+        }
+    }
+
+
+    public function checkPaths()
+    {
+        $config = MODX_CORE_PATH . 'config/' . MODX_CONFIG_KEY . '.inc.php';
+        if (is_writable($config)) {
+            if (@fileperms($config) & 0x0002) {
+                $this->warnings[$this->modx->lexicon('configcheck_configinc')] = $this->modx->lexicon('configcheck_configinc_msg', [
+                    'path' => $config,
+                ]);
+            }
+        }
+
+        $setup = MODX_BASE_PATH . 'setup/';
+        if (is_dir($setup)) {
+            $this->warnings[$this->modx->lexicon('configcheck_installer')] = $this->modx->lexicon('configcheck_installer_msg', [
+                'path' => $setup,
+            ]);
+        }
+
+        $cache = $this->modx->getCachePath();
+        if (!is_writable($cache)) {
+            $this->warnings[$this->modx->lexicon('configcheck_cache')] = $this->modx->lexicon('configcheck_cache_msg');
+        }
+    }
+
+
+    public function checkResources()
+    {
+        $auth = $this->getResource($this->modx->getOption('unauthorized_page', null, 1));
+        if (!$auth) {
+            $this->warnings[$this->modx->lexicon('configcheck_unauthorizedpage_unavailable')] =
+                $this->modx->lexicon('configcheck_unauthorizedpage_unavailable_msg');
+        } elseif (!$auth['published']) {
+            $this->warnings[$this->modx->lexicon('configcheck_unauthorizedpage_unpublished')] =
+                $this->modx->lexicon('configcheck_unauthorizedpage_unpublished_msg');
+        }
+
+        $err = $this->getResource($this->modx->getOption('error_page', null, 1));
+        if (!$err) {
+            $this->warnings[$this->modx->lexicon('configcheck_errorpage_unavailable')] =
+                $this->modx->lexicon('configcheck_errorpage_unavailable_msg');
+        } elseif (!$err['published']) {
+            $this->warnings[$this->modx->lexicon('configcheck_errorpage_unpublished')] =
+                $this->modx->lexicon('configcheck_errorpage_unpublished_msg');
+        }
+    }
+
+
+    public function checkSettings()
+    {
+
+        $tags = $this->modx->getCount('modSystemSetting', [
+            'key' => 'allow_tags_in_post',
+            'value' => 1,
+        ]);
+        if ($tags) {
+            $this->warnings[$this->modx->lexicon('configcheck_allowtagsinpost_system_enabled')] =
+                $this->modx->lexicon('configcheck_allowtagsinpost_system_enabled_msg');
+        }
+
+        $tags = $this->modx->getCount('modContextSetting', [
+            'context_key:!=' => 'mgr',
+            'key' => 'allow_tags_in_post',
+            'value' => 1,
+        ]);
+        if ($tags) {
+            $this->warnings[$this->modx->lexicon('configcheck_allowtagsinpost_context_enabled')] =
+                $this->modx->lexicon('configcheck_allowtagsinpost_context_enabled_msg');
+        }
+    }
+
+}
+
+return 'modSystemConfigCheck';
```

Comment:
We can use `modSystemConfigCheck::class` since 5.6 instead of strings. PHPStorm may show error when any mistake was done in the name accidentally (very rare case but it may happen).

Reply:
Nice!